File: 0020-valid-parentheses/0020-valid-parentheses.cpp
class Solution {
public:
    bool isValid(string s) {
        stack<char> s1;
        
        for(int i=0;i<s.length();i++){
            
            if(s[i]=='('||s[i]=='{'||s[i]=='['){
                s1.push(s[i]);
            }else if(s[i]==')'){
                if(s1.empty()||s1.top()!='(')return false;
                s1.pop();
            }else if(s[i]=='}'){
                if(s1.empty()||s1.top()!='{')return false;
                s1.pop();
            }else if(s[i]==']'){
                if(s1.empty()||s1.top()!='[')return false;
                s1.pop();
            }
        }
        if(s1.empty())return true;
        
        return false;
        
        
        
    }
};

File: 0020-valid-parentheses/0020-valid-parentheses.java
class Solution {
    public boolean isValid(String s) {
        Stack<Character> stack = new Stack<Character>();
        
        for(int i=0;i<s.length();i++){
            
            if(s.charAt(i)=='(' || s.charAt(i)=='{' || s.charAt(i)=='[' ){
                stack.push(s.charAt(i));
            }else{
                if(stack.isEmpty())return false;
                if(s.charAt(i)==')'){
                    if(stack.peek()!='('){
                        return false;
                    }
                }else if(s.charAt(i)=='}'){
                    if(stack.peek()!='{'){
                        return false;
                    }
                }else{
                    if(stack.peek()!='['){
                        return false;
                    }
                }
                stack.pop();
            }
        }
        if(!stack.isEmpty())return false;
        return true;
    }
}